### PR TITLE
docs: complete output config

### PR DESCRIPTION
**Pull request by SyMind**
## Summary

Complete the documentation of output configs.

## Checklist

- [ ] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: website/docs/en/config/output.mdx [hunk 1]
```diff
@@ -9,6 +9,38 @@ Used to indicate how and where Rspack outputs the contents of the generated file
...
+
+- **Type:** `string`
+
+:::warning
+Please use [`output.library.amdContainer`](#outputlibraryamdcontainer) instead as we might drop support for `output.amdContainer` in the future.
+:::
+
+Use a container(defined in global space) for calling `define`/`require` functions in an AMD module.
+
+:::warning
```
Comment (chenjiahan) [inline website/docs/en/config/output.mdx:22]: Can we merge to two warning blocks? 

Too many warning blocks is not good for reading, we can merge them into a list:

```md
:::warning

- Please use [`output.library.amdContainer`](#outputlibraryamdcontainer) instead as we might drop support for `output.amdContainer` in the future.
- Note that the value of `amdContainer` **must be** set as a global variable.

:::
```

File: website/docs/en/config/output.mdx [hunk 2]
```diff
@@ -1324,3 +1495,19 @@ Customize the main hot update filename. `[fullhash]` and `[runtime]` are availab
 - **Default:** `"webpackHotUpdate" + output.uniqueName`
 
 Only used when [chunkLoading](/config/output#outputchunkloading) is `"jsonp"` or `"import-scripts"`. This global variable is used for loading hot update chunks.
+
+## output.iife
+
+- **Type:** `boolean`
+- **Default:** `true`
+
+Tells webpack to add IIFE wrapper around emitted code.
```
Comment (chenjiahan) [inline website/docs/en/config/output.mdx:1504]:
```suggestion
Tells Rspack to add IIFE wrapper around emitted code.
```

File: website/docs/en/config/output.mdx [hunk 3]
```diff
@@ -1283,6 +1438,22 @@ The default value can be affected by different target:
 - Defaults to `'fetch'` if target is set to `'web'`, `'webworker'`, `'electron-renderer'` or `'node-webkit'`.
 - Defaults to `'async-node'` if target is set to `'node'`, `'async-node'`, `'electron-main'` or `'electron-preload'`.
 
+## output.webassemblyModuleFilename
+
+- **Type:** `string`
+- **Default:** `'[hash].module.wasm'`
+
+Specifies the filename of WebAssembly modules. It should be provided as a relative path within the `output.path` directory
```
Comment (chenjiahan) [inline website/docs/en/config/output.mdx:1446]:
```suggestion
Specifies the filename of WebAssembly modules. It should be provided as a relative path within the `output.path` directory.
```

File: website/docs/en/config/output.mdx [hunk 4]
```diff
@@ -364,13 +427,52 @@ import './publicPath.js';
...
+
+It is disabled by default (no compatibility, script URLs are strings).
+
+The policy name is taken from the object's `policyName` property.
+
+```js title="rspack.config.js"
+module.exports = {
+  //...
+  output: {
+    scriptType: 'module',
```
Comment (chenjiahan) [inline website/docs/en/config/output.mdx:471]: Incorrect example:

image

File: website/docs/en/config/output.mdx [hunk 5]
```diff
@@ -518,16 +614,43 @@ module.exports = {
...
+
+## output.hashFunction
+
+- **Type:** `'md4' | 'xxhash64'`
+- **Default:** `'md4'`
+
+The hashing algorithm to use.
+
+:::tip
+`hashFunction` supports `xxhash64` as a faster algorithm, which will be used as default when `experiments.futureDefaults` is enabled.
```
Comment (chenjiahan) [inline website/docs/en/config/output.mdx:639]: It seems like that Rspack does not supports `futureDefaults`.

Maybe we can enable `xxhash64` by default in v1.0

Comment (SyMind) [reply website/docs/en/config/output.mdx:639]: Rspack supports `futureDefaults`. I will include this feature in docs https://github.com/web-infra-dev/rspack/pull/6823/files .

**Review by chenjiahan — APPROVED** [review]
LGTM 👍

**Pull request merged**